### DISCUSSION TITLE
Trying to add Turbo to my monorepo but i get failing build on Vercel

### DISCUSSION THREAD

**Discussion by scerelli** [vercel/community#686] [Help]
Hi,
I tried to add turborepo on my project and locally everything works fine but on Vercel, at build time, i get typescript errors, sometimes I get an error in a piece of code sometimes in another but the thing they have in common is that the errors  are always related to the server side code.

Locally if i run 
`npx turbo run build --scope=ecommerce --filter=ecommerce...  ` it build correctly.

the same command ran on vercel fails, this is the most frequent error:

<img width="852" alt="image" src="https://user-images.githubusercontent.com/4821190/176130082-82d45f01-8e23-49ce-ab18-4ab091fb4b9d.png">

locally this error never showed UP and the `res` object is actually the next res object that comes in as a param of the endpoint function, so i don't really know what's happening on Vercel. 

this is my turbo settings

```
{
  "$schema": "https://turborepo.org/schema.json",
  "baseBranch": "origin/main",
  "remoteCache": {
    "signature": true
  },
  "pipeline": {
    "dev": {
      "cache": false
    },
    "build": {
      "dependsOn": ["^build"]
    },
    "cms#build": {
      "dependsOn": ["^build"],
      "outputs": [
        "dist/**"
      ]
    },
    "ecommerce#build": {
      "dependsOn": [
        "^build"
      ],
      "outputs": [
        ".next/**"
      ]
    },
    "lint": {
      "outputs": []
    }
  }
}
```

I hope you guys have an idea about what's going on here, thanks a lot

**Comment by swarnava** [comment]
Do you have a sample repository which we can use to reproduce?

**Reply by scerelli** [reply]
no i don't. 

the problem is that running the same exact command locally and while deploying gives me different results.

 i get type errors on vercel while locally it's all good. I don't understand why because the type are also different on vercel than what i see locally.
